### PR TITLE
Fixup how LS groups work.

### LINKED ISSUE #162: LipSync groups violating RFC5888
I. Section 5.3.1:

   The next step is to generate lip sync groups as defined in [RFC5888],
   Section 7.  For each MediaStream with more than one MediaStreamTrack,
   a group of type "LS" MUST be added that contains the mid values for
   each MediaStreamTrack in that MediaStream.  In some cases this may
   result in adding a mid to a given LS group that was not in that LS
   group in the associated offer.  Although this is not allowed by
   [RFC5888], it is allowed when implementing this specification.

So, do I understand it correctly the violation this paragraph is discussing is the following from Section 9.2 of RFC5888:

   The
   identification-tags contained in this "a=group" line MUST be the same
   as those received in the offer, or a subset of them (zero
   identification-tags is a valid subset).

So when will these violations occur for a LS group in JSEP. I assume due to that offer and answer has different structure of how MediaStreams uses the different m= lines. But, isn't this just another case of when an answerer will have to do an follow up Offer to be able to add all the MediaStreamTracks and their MediaStreams due to different sets on the two peers?

I don't really think this specification should override RFC 5888. The obvious solution is to make a new offer by the answering part who couldn't express its desired configuration in the first round.

### PR DESCRIPTION
**Pull request by fluffy**
Fixes #162 

Replaces https://github.com/rtcweb-wg/jsep/pull/167

### CODE REVIEW

**Comment by fluffy** [comment]
@juberti this just flattens all the changes in last PR to the final thin and puts it onto new base with no merge issue. Still needs changes to align with new sender / receiver objects

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -2621,19 +2621,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    target="RFC5888" />, it is allowed when implementing
-                    this specification. [[OPEN ISSUE: This is still
-                    under discussion. See:
-                    https://github.com/rtcweb-wg/jsep/issues/162.]]</t>
+                    <t>The next step is to generate session-level lip
+                    sync groups as defined in <xref target="RFC5888"/>,
+                    Section 7. All Tracks in the same MediaStream need
+                    to be placed in the same "LS" group as long as
+                    theses tracks were in the offered to by synchronized
+                    in the offer.  RFC5888 only allows the groups in the
```
Comment (alvestrand) [inline draft-ietf-rtcweb-jsep.xml]: I can't parse the "as long as" sentence. The sentence seems to make sense if it stops with "in the same "LS" group". If the rest of the sentence modifies it, it's not clear what it says.

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -2621,19 +2621,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                    answer to be a subset of what was in the offer. If
+                    there are tracks that the same MediaStream that can
+                    not be added to the "LS" group, once the state moves
+                    back to "stable" the browser MAY initiate another
+                    offer/answer negotiation put all the tracks in the
+                    MediaStream into the "LS" group or they can be left
+                    in a state that though answer wishes them to be
+                    synchronized, the answer can not signal them as
+                    synchronized because the offer did put them in the
+                    same LS group. The working group consider varios
```
Comment (alvestrand) [inline draft-ietf-rtcweb-jsep.xml]: English nit: "The working group considered various solutions to this rare corner condition". In general, I don't think WG considerations belong in the normative part of the document, so I'd be fine with yanking the whole sentence.

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -2621,19 +2621,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                    theses tracks were in the offered to by synchronized
+                    in the offer.  RFC5888 only allows the groups in the
+                    answer to be a subset of what was in the offer. If
+                    there are tracks that the same MediaStream that can
+                    not be added to the "LS" group, once the state moves
+                    back to "stable" the browser MAY initiate another
+                    offer/answer negotiation put all the tracks in the
+                    MediaStream into the "LS" group or they can be left
+                    in a state that though answer wishes them to be
+                    synchronized, the answer can not signal them as
```
Comment (alvestrand) [inline draft-ietf-rtcweb-jsep.xml]: This paragraph is hard to parse. The paragraph seems to say the same thing twice in two different ways, but the language is different enough that I can't be sure there's not a subtle difference.

Suggested replacement for the whole paragraph starting at "If there are tracks... if it is critical for their use case": "If there are tracks that belong to a MediaStream that cannot be added to the "LS" group defined by the offerer, the answerer MAY add them without synchronization. It MAY also skip adding the tracks to the answer, and instead create a new offer after sending the answer."

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -2621,19 +2621,12 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    not in that LS group in the associated
-                    offer. Although this is not allowed by <xref
-                    target="RFC5888" />, it is allowed when implementing
-                    this specification. [[OPEN ISSUE: This is still
-                    under discussion. See:
-                    https://github.com/rtcweb-wg/jsep/issues/162.]]</t>
+                    <t>The next step is to generate session-level lip
+                    sync groups as defined in <xref target="RFC5888"/>,
+                    Section 7. For each group of type "LS" present in the offer, determine which of the RtpTransceivers referenced
+                    by the supplied mid values reference a common MediaStream.
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: a common MediaStream in the answer?

**Comment by juberti** [comment]
PTAL

**Comment by fluffy** [comment]
LGTM

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -2621,19 +2621,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    offer. Although this is not allowed by <xref
-                    target="RFC5888" />, it is allowed when implementing
-                    this specification. [[OPEN ISSUE: This is still
-                    under discussion. See:
-                    https://github.com/rtcweb-wg/jsep/issues/162.]]</t>
+                    <t>The next step is to generate session-level lip
+                    sync groups as defined in <xref target="RFC5888"/>,
+                    Section 7. For each group of type "LS" present in the offer,
+                    determine which of the local RtpTransceivers identified by
+                    the group's mid values reference a common local MediaStream
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Nit: perhaps "that group's" mid values?

is which singular here so we need "references"? not sure.

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -2621,19 +2621,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+                    (as specified in the addTrack and
+                    addTransceiver methods). If at least two such
+                    RtpTransceivers exist, a group of type "LS" with the mid
+                    values of these RtpTransceivers MUST be added. Otherwise,
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Nit: "those"?

**Comment by ekr** [comment]
LGTM

File: draft-ietf-rtcweb-jsep.xml (hunk 6 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Since 'these' references something nearby, and the transceivers were mentioned in the initial clause, I'm inclined to keep 'these'.

Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Fine by me. The question mark indicated I was unsure.

File: draft-ietf-rtcweb-jsep.xml (hunk 5 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: changed to 'that'; agree that's slightly more specific
which is plural here, I believe
